components/automate-ui/src/app/pages/user-details/user-details.component.ts

```diff
@@ -65,18 +66,35 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
         this.user = { ...user };
       });
 
-    this.route.data.subscribe((data: { isNonAdmin: boolean }) => {
-      this.isAdminView = !data.isNonAdmin;
-      // Update the forms once this info has come in
-      this.createForms(this.fb);
-      this.resetForms();
-    });
+    this.route.data.pipe(
+      takeUntil(this.isDestroyed))
+      .subscribe((data: { isNonAdmin: boolean }) => {
+        this.isAdminView = !data.isNonAdmin;
+        // Update the forms once this info has come in
+        this.createForms(this.fb);
+        this.resetForms();
+      });
 
     this.store.select(updateStatus).pipe(
       takeUntil(this.isDestroyed),
-      filter(status => status === EntityStatus.loadingSuccess))
-      // same status is used for updating password or full name, so we just reset both
-      .subscribe(() => this.resetForms());
+      filter(status => this.updatingUser && !loading(status)))
+      .subscribe(() => {
+        this.updatingUser = false;
```

Comment:
added `updatingUser` for consistency with our pages and to distinguish from `deletingUser`